File: PY2C++/src/ProjectConverter/ProgramConverter.cpp
#include <iostream>
#include <cstdlib>
#include "Token.cpp"
#include <list>
#include <iterator>
#include "Scanner.cpp"

using namespace std;
bool hadError = false;

void report(int line, string where, string message){
    cout << "[line " << line << " ] Error" << where << ": " << message;
    hadError = true;
}

void error(int line, string message){
    report(line, "", message);
}

void run(string source){
    Scanner scanner = new Scanner(source);
    list <Token> tokens = scanner.scanTokens();
    for (Token token : tokens){
        token.print() ;
        cout <<" \n";
    }
}

void runFile(string path){
    // Don't know still :(
}

void runPrompt(){
    for(;;){
        string s;
        cout <<"> ";
        cin >> s;
        run(s);
        hadError = false;
    }
}

int main(int args, char* argv[]){
    if (args > 1){
        cout <<" Usage: programConverter [script] \n";
        exit(0);
    }
    else if (args == 1){
        runFile(argv[0]);
    }
    else {
        runPrompt();
    }
}

File: PY2JS/src/ProgramConverter.mjs
import fs from 'fs'
import { scanTokens } from './Scanner.mjs'

let hadError = false

const runFile = params => {
    fs.readFile(params, 'utf-8', (err, data) => {
        if (err) process.exit(65)
        run(data)
    })
    if (hadError) process.exit(65)
}

const runPrompt = () => {
    for (; ;) {
        console.log('>')
        process.stdin.resume();
        process.stdin.setEncoding("utf-8");
        let stdin_input = "";
        process.stdin.on("data", function (input) {
            stdin_input = input;
        });
        process.stdin.on("end", function () {
            run(stdin_input);
        });
        hadError = false
    }
}

const run = params => {
    let tokens = scanTokens(params)
    tokens.forEach(element => {
        console.log(element)
    });
}

export const error = (line, message) => {
    report(line, "", message)
}

const report = (line, where, message) => {
    console.error("[line " + line + " ] Error" + where + ": " + message)
    hadError = true
}

if (process.argv.length > 3) {
    console.log('Usage: programConverter [script]')
    process.exit(64)
} else if (process.argv.length === 3) {
    runFile(process.argv[2])
} else {
    runPrompt()
}
